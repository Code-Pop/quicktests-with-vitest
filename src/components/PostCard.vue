<template>
    <div>
        <h1>{{ title }}</h1>
        <p>{{ body }}</p>
    </div>
</template>

<script setup>
defineProps({
    title: {
        type: String,
        required: true,
    },
    body: {
        type: String,
        required: true,
    },
});
</script>

<style>
h1 {
    font-size: 32px;
    font-weight: 700;
    color: #fff;
}
</style>