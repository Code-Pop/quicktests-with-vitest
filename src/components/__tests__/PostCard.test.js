/* eslint-disable no-undef */
import { mount } from '@vue/test-utils'

import PostCard from '../PostCard.vue'
import { expect } from 'vitest'

describe('Post Card Component', () => {
  test('created posts render correctly', () => {
    const title = 'Test Post'
    const body = 'test post body...'
    const wrapper = mount(PostCard, {
      props: { title, body }
    })

    expect(wrapper.html()).toMatchSnapshot()
  })
})
