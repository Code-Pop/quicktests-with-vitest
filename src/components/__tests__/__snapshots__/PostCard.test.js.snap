// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Post Card Component > created posts render correctly 1`] = `
"<div>
  <h1>Test Post</h1>
  <p>test post body...</p>
</div>"
`;
